File: aifupan_project/fupan-server/replay-words/src/main/java/com/jiuyu/replay/words/bo/BasicSettingsBo.java
package com.jiuyu.replay.words.bo;

import com.jiuyu.replay.common.constant.WordsEnum;
import com.jiuyu.replay.generic.dto.words.BasicSettingsBaseDto;
import com.jiuyu.replay.words.bo.anchor.AddOrUpdateAnchorBo;
import com.jiuyu.replay.words.bo.video.UpdateAiPartialBo;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * @author ：lujie
 * @description：基础设置业务对象
 * @date ：2025/1/7
 */
@Data
@Schema(description = "基础设置业务对象")
@AllArgsConstructor
@NoArgsConstructor
public class BasicSettingsBo extends BasicSettingsBaseDto {

    @Schema(description = "主键")
    private Long id;

    @Schema(description = "来源id")
    @NotNull(message = "来源id不能为空")
    private String sourceId;

    @Schema(description = "来源类型 0：主播，1：视频，2：文件，")
    @NotNull(message = "来源类型不能为空")
    private Integer sourceType;

    @Schema(description = "用户表id")
    private Long userId;

    @Schema(description = "租户id")
    private Long tenantId;

    @Schema(description = "创建时间")
    private Date createDate;

    @Schema(description = "更新时间")
    private Date updateDate;

    public BasicSettingsBo(AddOrUpdateAnchorBo bo) {
        this.userId = bo.getUserId();
        this.tenantId = bo.getTenantId();
        this.sourceType = WordsEnum.basicSettingsType.ANCHOR.getCode();
        this.sourceId = bo.getSecUid();

        // 设置值
        setValues(bo);
    }

    public BasicSettingsBo(UpdateAiPartialBo bo) {
        this.userId = bo.getUserId();
        this.tenantId = bo.getTenantId();
        this.sourceType = WordsEnum.basicSettingsType.ANCHOR.getCode();
        this.sourceId = bo.getSecUid();
        this.accountType = bo.getAccountType();
        this.accountStage = bo.getAccountStage();
        this.accountWaterLevel = bo.getAccountWaterLevel();
        this.accountFlow = bo.getAccountFlow();
        this.anchorSituation = bo.getAnchorSituation();
        this.roiAccuracy = bo.getRoiAccuracy();
    }

    public <T extends BasicSettingsBaseDto> void setValues(T bo) {
        this.accountType = bo.getAccountType();
        this.premiereDate = bo.getPremiereDate();
        this.accountStage = bo.getAccountStage();
        this.accountWaterLevel = bo.getAccountWaterLevel();
        this.accountFlow = bo.getAccountFlow();
        this.livingTarget = bo.getLivingTarget();
        this.livingModality = bo.getLivingModality();
        this.marketing = bo.getMarketing();
        this.optimizeDirection = bo.getOptimizeDirection();
        this.learning = bo.getLearning();
        this.livingMode = bo.getLivingMode();
        this.anchorSituation = bo.getAnchorSituation();
        this.roiAccuracy = bo.getRoiAccuracy();
    }

}

File: aifupan_project/ReviewAnalysis/bo/BasicSettingsBo.cs
namespace ReviewAnalysis.bo
{
    public class BasicSettingsBo
    {
        /// <summary>
        /// 账号归属类型 0：自有账号 1：同行账号
        /// </summary>
        public int? accountType { get; set; }
        
        /// <summary>
        /// 首播日期
        /// </summary>
        public string premiereDate { get; set; }
        
        /// <summary>
        /// 账号阶段，使用字典account_stage的值
        /// </summary>
        public int? accountStage { get; set; }
        
        /// <summary>
        /// 账号水平， 使用字典account_water_level的值
        /// </summary>
        public int? accountWaterLevel { get; set; }

        /// <summary>
        /// 浏览结构，使用字典 account_flow的值
        /// </summary>
        public int? accountFlow { get; set; }
        
        /// <summary>
        /// 直播目标，使用字典living_target的值
        /// </summary>
        public int? livingTarget { get; set; }
        
        /// <summary>
        /// 直播方式，使用字典living_modality的值
        /// </summary>
        public int? livingModality { get; set; }
        
        /// <summary>
        /// 营销方式，使用字典marketing的值
        /// </summary>
        public int? marketing { get; set; }
        
        /// <summary>
        /// 优化方向，使用字典optimize_direction的值
        /// </summary>
        public string optimizeDirection { get; set; }
        
        /// <summary>
        /// 学习方向，使用字典learning的值
        /// </summary>
        public string learning { get; set; }
        
        /// <summary>
        /// 直播模式，使用字典living_mode的值
        /// </summary>
        public int? livingMode { get; set; }
        
        /// <summary>
        /// 主播账号情况描述
        /// </summary>
        public string anchorSituation { get; set; }

        /// <summary>
        /// ROI观测精度，使用字典roi_accuracy的值
        /// </summary>
        public string roiAccuracy { get; set; }
    }
}